app/code/community/Inchoo/SocialConnect/Block/Google/Button.php

```diff
@@ -35,16 +40,26 @@ protected function _construct() {
 
         $this->userInfo = Mage::registry('inchoo_socialconnect_google_userinfo');
 
-        // CSRF protection
-        Mage::getSingleton('core/session')->setGoogleCsrf($csrf = md5(uniqid(rand(), true)));
-        $this->client->setState($csrf);
-
         Mage::getSingleton('customer/session')
             ->setSocialConnectRedirect(Mage::helper('core/url')->getCurrentUrl());
 
         $this->setTemplate('inchoo/socialconnect/google/button.phtml');
     }
 
+
+    protected function _beforeToHtml()
+    {
+        if (!static::$csrf) {
+            static::$csrf = Mage::getSingleton('core/session')->getGoogleCsrf();
```

Comment:
Err... `google_csrf` is not set on the session yet. Shouldn't that be `md5(uniqid(rand(), true))`?